homeassistant/components/lutron/__init__.py

```diff
@@ -145,8 +145,15 @@ def button_callback(self, button, context, event, params):
             # A single-action button; the Lutron controller won't tell us
             # when the button is released, so use a different action name
             # than for buttons where we expect a release event.
-            action = 'single'
-
-        data = {ATTR_ID: self._id, ATTR_ACTION: action}
+            #
+            # Some Lutron keypads lie about their button types, though,
+            # and send release notifications even for single-action
+            # buttons. So we need to check that this is indeed a press.
+            if event == Button.Event.PRESSED:
+                action = 'single'
+            else:
+                action = None
```

Comment:
This logic can be simplified.  Move `action = None` above the first `if` block, then simplify the else condition to `elif event == Button.Event.PRESSED`...